javacord-core/src/main/java/org/javacord/core/util/handler/guild/GuildEmojisUpdateHandler.java

```diff
@@ -87,16 +88,15 @@ public void handle(JsonNode packet) {
             });
 
             Set<Long> emojiIds = emojis.keySet();
-            server.getCustomEmojis().stream()
+            Set<KnownCustomEmoji> emojiToDelete = server.getCustomEmojis().stream()
                     .filter(emoji -> !emojiIds.contains(emoji.getId()))
-                    .forEach(emoji -> {
-                        api.removeCustomEmoji(emoji);
-                        server.removeCustomEmoji(emoji);
-
-                        KnownCustomEmojiDeleteEvent event = new KnownCustomEmojiDeleteEventImpl(emoji);
-
-                        api.getEventDispatcher().dispatchKnownCustomEmojiDeleteEvent(server, emoji, server, event);
-                    });
+                    .collect(Collectors.toSet());
+            emojiToDelete.forEach(emoji -> {
+                api.removeCustomEmoji(emoji);
+                server.removeCustomEmoji(emoji);
+                KnownCustomEmojiDeleteEvent event = new KnownCustomEmojiDeleteEventImpl(emoji);
+                api.getEventDispatcher().dispatchKnownCustomEmojiDeleteEvent(server, emoji, server, event);
```

Comment:
event is only used once so just instantiate in dispatch function

```suggestion           

api.getEventDispatcher().dispatchKnownCustomEmojiDeleteEvent(server, emoji, server, 
new KnownCustomEmojiDeleteEventImpl(emoji));
```